src/dynarec/rv64/dynarec_rv64_df.c

```diff
@@ -123,6 +123,8 @@ uintptr_t dynarec64_DF(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t ip, int ni
                         FRFLAGS(x5);   // get back FPSR to check the IOC bit
                         ANDI(x5, x5, 1<<FR_NV);
                         CBZ_NEXT(x5);
+                        SRLIW(x5, x4, 16);
```

Comment:
You are sure this will work for negative values. I think it will never be 0 because the high part will be filled with 0xffffffff after the `FCVTWD`?

Reply:
Sorry, I've force-pushed, I believe it's right this time.